buildconf

```diff
@@ -99,6 +94,49 @@ else
   autoconf_flags="-f"
 fi
 
+# Check PHP build system tools such as autoconf and their versions.
+if ! test -f "$stamp"; then
```

Comment:
These checks all look very cheap to me, is it really worthwhile to ensure they're only run once?

Reply:
Performance here is happening on the micro optimization level.

With this stamp file check:

```bash
time buildconf
6.569sec total
# here nothing is done on the build system files
time buildconf
0.056sec total
# here some config.m4 file is changed
time buildconf
6-7sec total (again)
```

Without this check:

```bash
time buildconf
6.749sec total
# here nothing is done on the build system files
time buildconf
0.135sec total
# here some config.m4 file is changed
time buildconf
6-7sec total (again)
```

Here the difference is between those 0.056sec vs 0.135sec.

It was done when probably there was more than a second or two difference and more tools checked:
https://github.com/php/php-src/blob/aa5f0bdfd28771944ecf137e372a7cea1714f02f/build/buildcheck.sh

The check of the tools and version is ok and helps actually to know what to install, but no one will realistically notice the performance implication here if we remove this timestamp file check. That's for sure. The numbers are the same when doing this on the build.mk file level also... It is 2019 after all also, yes, with better hardware and everything. I agree.

Remove the stamp file check?

Reply:
Thanks for checking. Given those numbers, yeah, definitely in favor of dropping the stamp file. Especially if you take into account that you're going to be running configure after that, which takes a lot of time...